src/libraries/System.Drawing.Common/src/System/Drawing/Imaging/ColorMatrix.cs

```diff
@@ -406,7 +406,7 @@ internal static unsafe class PinningMarshaller
             public static ref float GetPinnableReference(ColorMatrix managed) => ref (managed is null ? ref Unsafe.NullRef<float>() : ref managed.GetPinnableReference());
 
             // All usages in our currently supported scenarios will always go through GetPinnableReference
-            public static float* ConvertToUnmanaged(ColorMatrix managed) => throw new UnreachableException();
+            public static float* ConvertToUnmanaged(ColorMatrix _) => throw new UnreachableException();
```

Comment:
@jkoritzinsky, if in such situations we statically know these methods will never be used, can we update the generator to be ok with them not existing, and then just delete them rather than having these `throw new UnreachableException()` dummy implementations?

Reply:
There was a lot of desire on the interop team to always require specific methods for particular shapes to reduce the concept count we'd have to teach users. cc: @AaronRobinsonMSFT since he felt the most strongly about it.

Reply:
> can we update the generator to be ok with them not existing, and then just delete them

This breaks the interop contract and is something we are being very explicit about going forward. The intent here is to avoid at all costs magic from the source generator scenarios. We expect functions (A), (B), and (C), if that shape isn't there then it is a broken interop contract.

Reply:
Understood.  I'm saying today it is:

- "We expect functions (A), (B), and (C), if that shape isn't there then it is a broken interop contract.

but it could be:

- "If (A) is available, we'll use A.  If (B) is available, we'll use B.  If (C) is available, we'll use C.".

The C# compiler does this in many places, for example.

Reply:
> The C# compiler does this in many places, for example.

Can you share an example?

Reply:
> Can you share an example?

e.g.
- If you foreach something that implements IDisposable, the code will be emitted with a try/finally calling Dispose; if it doesn't, no try/finally, no Dispose call.
- When calling `M(params T[] args)` with no args, it'll use `Array.Empty<T>()` if it's available, or else `new T[0]`.
- When emitting code for `$"Hello, {name}"`, it'll use `DefaultInterpolatedStringBuilder` if available, otherwise it'll use `string.Format`.
- When emitting code for await, if the thing you're awaiting has `UnsafeOnCompleted` it'll use it, otherwise it'll use `OnCompleted`.
...